src/test/java/org/broadinstitute/hellbender/utils/gcs/BucketUtilsTest.java

```diff
@@ -31,22 +32,36 @@
     }
 
     @Test(groups={"bucket"})
-    public void testIsCloudStorageURL(){
-        Assert.assertTrue(BucketUtils.isCloudStorageUrl("gs://abucket/bucket"));
-        Assert.assertFalse(BucketUtils.isCloudStorageUrl("hdfs://namenode/path/to/file"));
-        Assert.assertFalse(BucketUtils.isCloudStorageUrl("localFile"));
+    public void testIsGcsUrl(){
+        Assert.assertTrue(BucketUtils.isGcsUrl("gs://abucket/bucket"));
+        Assert.assertFalse(BucketUtils.isGcsUrl("hdfs://namenode/path/to/file"));
+        Assert.assertFalse(BucketUtils.isGcsUrl("localFile"));
+        Assert.assertFalse(BucketUtils.isGcsUrl("https://www.somewhere.com"));
+    }
 
-        Assert.assertTrue(BucketUtils.isCloudStorageUrl(Paths.get(URI.create("gs://abucket/bucket"))));
-        // We cannot run this one because the HDFS provider looks for the "namenode" machine
-        // and throws an exception.
-        //Assert.assertFalse(BucketUtils.isCloudStorageUrl(Paths.get(URI.create("hdfs://namenode/path/to/file"))));
-        Assert.assertFalse(BucketUtils.isCloudStorageUrl(Paths.get("localFile")));
+    @DataProvider
+    public Object[][] getVariousPathsForPrefetching(){
+        return new Object[][]{
+                {"localFile", false},
+                {"files:///local/file", false},
+                {"http://www.somewhere.com", true},
+                {"https://www.somewhere.com", true},
+               // {"hdfs://namenode/path/to/file", false}, skip this because it fails when namenode doesn't exist
```

Comment:
You could unit test the HDFS case by calling the overload that takes the scheme as a raw String...

Reply:
It seems to work with the GATKPathSpecifier so I added a specific test for it.